src/core/io/platform/f4xx/ADCf4xx.cpp

```diff
@@ -84,109 +127,222 @@ void ADCf4xx::initADC(uint8_t num_channels) {
+        __HAL_RCC_ADC3_CLK_ENABLE();
+        break;
+    default:
+        return; // Should never get here
+    }
+    HAL_ADC_Init(halADC);
 }
 
 void ADCf4xx::initDMA() {
-    halDMA.Instance                 = DMA2_Stream0;
-    halDMA.Init.Direction           = DMA_PERIPH_TO_MEMORY;
-    halDMA.Init.PeriphInc           = DMA_PINC_DISABLE;
-    halDMA.Init.MemInc              = DMA_MINC_ENABLE;
-    halDMA.Init.PeriphDataAlignment = DMA_PDATAALIGN_HALFWORD;
-    halDMA.Init.MemDataAlignment    = DMA_MDATAALIGN_HALFWORD;
-    halDMA.Init.Mode                = DMA_CIRCULAR;
-    halDMA.Init.Priority            = DMA_PRIORITY_VERY_HIGH;
-    halDMA.Init.FIFOMode            = DMA_FIFOMODE_DISABLE;
-
-    HAL_DMA_Init(&halDMA);
-    __HAL_LINKDMA(&halADC, DMA_Handle, halDMA);
+    uint8_t adcNum                 = getADCNum();
+    ADCf4xx::ADC_State_t* adcState = &adcArray[adcNum];
+    DMA_HandleTypeDef* dma         = &adcState->halDMA;
+    // Set DMA instance to proper config settings
+    switch (adcNum) {
+    case ADC1_SLOT:
+        dma->Instance     = DMA2_Stream0;
+        dma->Init.Channel = DMA_CHANNEL_0;
+        break;
+    case ADC2_SLOT:
+        dma->Instance     = DMA2_Stream2;
+        dma->Init.Channel = DMA_CHANNEL_1;
+        break;
+    case ADC3_SLOT:
+        dma->Instance     = DMA2_Stream1;
+        dma->Init.Channel = DMA_CHANNEL_2;
+        break;
+    default:
+        return; // Should never get here
+    }
+    dma->Init.Direction           = DMA_PERIPH_TO_MEMORY;
+    dma->Init.PeriphInc           = DMA_PINC_DISABLE;
+    dma->Init.MemInc              = DMA_MINC_ENABLE;
+    dma->Init.PeriphDataAlignment = DMA_PDATAALIGN_HALFWORD;
+    dma->Init.MemDataAlignment    = DMA_MDATAALIGN_HALFWORD;
+    dma->Init.Mode                = DMA_CIRCULAR;
+    dma->Init.Priority            = DMA_PRIORITY_VERY_HIGH;
+    dma->Init.FIFOMode            = DMA_FIFOMODE_DISABLE;
+    dma->Init.FIFOThreshold       = DMA_FIFO_THRESHOLD_FULL;
+    dma->Init.MemBurst            = DMA_MBURST_SINGLE;
+    dma->Init.PeriphBurst         = DMA_PBURST_SINGLE;
+
+    HAL_DMA_Init(dma);
+
+    switch (adcNum) {
+    case ADC1_SLOT:
+        HAL_NVIC_SetPriority(DMA2_Stream0_IRQn, platform::ADC_INTERRUPT_PRIORITY, 0);
+        HAL_NVIC_EnableIRQ(DMA2_Stream0_IRQn);
+        break;
+    case ADC2_SLOT:
+        HAL_NVIC_SetPriority(DMA2_Stream2_IRQn, platform::ADC_INTERRUPT_PRIORITY, 0);
+        HAL_NVIC_EnableIRQ(DMA2_Stream2_IRQn);
+        break;
+    case ADC3_SLOT:
+        HAL_NVIC_SetPriority(DMA2_Stream1_IRQn, platform::ADC_INTERRUPT_PRIORITY, 0);
+        HAL_NVIC_EnableIRQ(DMA2_Stream1_IRQn);
+        break;
+    default:
+        return; // Should never get here
+    }
+    __HAL_LINKDMA(&adcState->halADC, DMA_Handle, *dma);
 }
 
 void ADCf4xx::addChannel(uint8_t rank) {
     GPIO_InitTypeDef gpioInit;
-    Pin myPins[]      = {pin};
     uint8_t numOfPins = 1;
+    uint32_t channel;
+    Pin myPins[]                   = {pin};
```

Comment:
Get rid of `myPins`. That's a terrible variable name

Reply:
ahahaha I never noticed thats what it was called. Thats grandfathered in from f3 adc